src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/DiagnosticExtensions.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace System.Runtime.Analyzers
+{
+    internal static class DiagnosticExtensions
```

Comment:
All these seem like good helpers which are useful for all kind of analyzers. Probably someday we can make these public in CodeAnalysis layer and avoid duplication?